configs/components/facter.rb

```diff
@@ -226,6 +224,23 @@
     ["#{make} -j$(shell expr $(shell #{platform[:num_cores]}) + 1) install"]
   end
 
+  if platform.is_osx?
+    ldd = "otool -L"
+  else
+    ldd = "ldd"
+  end
+
+  unless platform.is_windows? || platform.is_cross_compiled_linux? || platform.architecture == 'sparc'
+    pkg.check do
+      [
+        # Check that we're not linking against system libstdc++ and libgcc_s
+        "#{ldd} lib/libfacter.so",
+        "[ $$(#{ldd} lib/libfacter.so | grep -c libstdc++) -eq 0 ] || #{ldd} lib/libfacter.so | grep libstdc++ | grep -v ' /lib'",
+        "[ $$(#{ldd} lib/libfacter.so | grep -c libgcc_s) -eq 0 ] || #{ldd} lib/libfacter.so | grep libgcc_s | grep -v ' /lib'"
```

Comment:
I'm not blocking on it, but whitespace specific grep's seem like they'll be fragile. Let's run with this for a while and see how it goes.